doc/modules/impute.rst

```diff
@@ -52,5 +52,36 @@ Note that, here, missing values are encoded by 0 and are thus implicitly stored
 in the matrix. This format is thus suitable when there are many more missing
 values than observed values.
 
-:class:`SimpleImputer` can be used in a Pipeline as a way to build a composite
-estimator that supports imputation. See :ref:`sphx_glr_auto_examples_plot_missing_values.py`.
+A more sophisticated approach is to use the :class:`MICEImputer` class, which
+implements the multivariate imputation by chained equations technique. While
+the :class:`SimpleImputer` class fills in each missing value in a feature
+column by a constant that is a function of that feature only, MICE fills in
+each feature's missing values based on the values of the other features as
+well. It does so in a round-robin fashion: at each step, a feature column is
+designated as output and the other feature columns are treated as inputs. A
+regressor is fit on the non-missing values of the output feature column and the
+corresponding values in the input columns. Then, the regressor is applied so as
+to predict the unknown values of the output feature. This is repeated for each
+feature, and then is done for a number of imputation rounds. Here is an example
+snippet::
+
+    >>> import numpy as np
+    >>> from sklearn.impute import MICEImputer
+    >>> imp = MICEImputer(n_imputations=10, random_state=0)
+    >>> imp.fit([[1, 2], [np.nan, 3], [7, np.nan]])
+    MICEImputer(imputation_order='ascending', initial_strategy='mean',
+          max_value=None, min_value=None, missing_values='NaN', n_burn_in=10,
+          n_imputations=10, n_nearest_features=None, predictor=None,
+          random_state=0, verbose=False)
+    >>> X_test = [[np.nan, 2], [6, np.nan], [np.nan, 6]]
+    >>> print(np.round(imp.transform(X_test)))            # doctest: +ELLIPSIS
+    [[ 1.  2.]
+     [ 6.  4.]
+     [13.  6.]]
+
+Note that MICE uses sampling as part of its imputation process, so the exact
+values of the imputed data will be slightly different each time you run it.
```

Comment:
"will be slightly different each time you run it": is that true even if you set `random_state`?

Reply:
What I mean is that that `A` and `B` will not be exactly equal in the code below:

```
mice = MICE(...).fit(X_train)
A = mice.impute(X_valid)
B = mice.impute(X_valid)
```

Reply:
I think @lesteve wants you to qualify your statement with "unless random_state is fixed" if that's true.

Reply:
This is probably the wrong place to discuss this anyway. I think I'll remove the note entirely.